src/Files.App/ViewModels/Properties/SecurityAdvancedViewModel.cs

```diff
@@ -0,0 +1,218 @@
+
+		public SecurityAdvancedViewModel(ListedItem item, Window window)
+		{
+			IsFolder = item.PrimaryItemAttribute == StorageItemTypes.Folder && !item.IsShortcut;
+			Item = item;
+			Window = window;
+
+			InitializeCommands();
+			GetAccessControlList();
+		}
+
+		public SecurityAdvancedViewModel(DriveItem item, Window window)
+		{
+			IsFolder = true;
+			Item = new ListedItem()
+			{
+				ItemNameRaw = item.Text,
+				ItemPath = item.Path,
+				PrimaryItemAttribute = StorageItemTypes.Folder
+			};
+			Window = window;
+
+			InitializeCommands();
+			GetAccessControlList();
+		}
+
+		private void InitializeCommands()
+		{
+			ChangeOwnerCommand = new RelayCommand(ChangeOwner, () => AccessControlList is not null);
+			AddAccessControlEntryCommand = new RelayCommand(AddAccessControlEntry, () => AccessControlList is not null && AccessControlList.IsValid);
+			RemoveAccessControlEntryCommand = new RelayCommand(RemoveAccessControlEntry, () => AccessControlList is not null && AccessControlList.IsValid && SelectedAccessControlEntry is not null);
+			DisableInheritanceCommand = new RelayCommand(DisableInheritance, () => AccessControlList is not null && AccessControlList.IsValid && (AccessControlList.IsProtected != _isProtected));
+			SetDisableInheritanceOptionCommand = new RelayCommand<string>(SetDisableInheritanceOption);
+			ReplaceChildPermissionsCommand = new RelayCommand(ReplaceChildPermissions, () => AccessControlList is not null && AccessControlList.IsValid);
+		}
+
+		private async void ChangeOwner()
+		{
+			var pickedObject = await OpenObjectPicker();
+			if (pickedObject is not null)
+			{
+				bool isFolder = Item.PrimaryItemAttribute == StorageItemTypes.Folder && !Item.IsShortcut;
+
+				// Set owner and refresh file permissions
+				if (FileOperationsHelpers.SetFileOwner(Item.ItemPath, pickedObject))
+					GetAccessControlList();
+			}
+		}
+
+		private async void AddAccessControlEntry()
+		{
+			// Pick an user/group
+			var sid = await OpenObjectPicker();
+			if (string.IsNullOrEmpty(sid))
+				return;
+
+			// Initialize
+			var ace = FileSecurityHelpers.InitializeDefaultAccessControlEntry(IsFolder, sid);
+			AccessControlList.AccessControlEntries.Add(ace);
+
+			// Save
+			FileSecurityHelpers.SetAccessControlList(AccessControlList);
+		}
+
+		private void RemoveAccessControlEntry()
+		{
+			// Remove
+			AccessControlList.AccessControlEntries.Remove(SelectedAccessControlEntry);
+
+			// Save
+			FileSecurityHelpers.SetAccessControlList(AccessControlList);
+		}
+
+		private void DisableInheritance()
+		{
+			// Update protection status and refresh access control
+			if (FileOperationsHelpers.SetAccessRuleProtection(Item.ItemPath, IsFolder, _isProtected, _preserveInheritance))
+				GetAccessControlList();
+		}
+
+		private void SetDisableInheritanceOption(string options)
+		{
+			_isProtected = bool.Parse(options.Split(',')[0]);
+			_preserveInheritance = bool.Parse(options.Split(',')[1]);
+
```

Comment:
It might be better to split `options` just once and make sure it has 2 items

Reply:
I avoid updates of security functionalities. Future PR will fix and I keep this in mind.